### ISSUE TITLE
Add Node operator: separate label

### ISSUE THREAD

**Issue by ghost** [Bforartists/Bforartists#2454]
this is really because Blender layout.operator doesn't have "icon_only" option, so we can't set text and show icon only at same time, also operator don't have dynamic label (like description)

so this task would be to add either of these options, or maybe there's another way I'm missing

examples:
In Toolshelf
![image](https://user-images.githubusercontent.com/25552173/119266164-01b03a80-bbea-11eb-8184-9ee8fd2a920a.png)

In Nodeshelf
![image](https://user-images.githubusercontent.com/25552173/119266032-6cad4180-bbe9-11eb-81dc-47fe136a9068.png)

**Comment by ReinerBforartists** [comment]
Ah, great. So still broken in the nodes editor ^^

No idea why it works just fine in the tool shelf though :)

**Comment by iyadahmed** [comment]
> Ah, great. So still broken in the nodes editor ^^
> 
> 
> 
> No idea why it works just fine in the tool shelf though :)

Hmm, the title is same for same operators, for example apply transform buttons (Location, Rotation, etc) in Apply panel

**Comment by ReinerBforartists** [comment]
I really thought your commit has fixed this issue. Seems that we need to continue with our investigation :)

**Comment by iyadahmed** [comment]
> I really thought your commit has fixed this issue. Seems that we need to continue with our investigation :)

Ah yes, this is a different topic, the commit just added ability for add node operator to display node description

**Comment by iyadahmed** [comment]
> Ah, great. So still broken in the nodes editor ^^
> 
> 
> 
> No idea why it works just fine in the tool shelf though :)

To be more clear the issue is in both toolshelf and nodeshelf

**Comment by ReinerBforartists** [comment]
For me it works in the toolshelf.

![worksfine](https://user-images.githubusercontent.com/13919702/119270744-6b3a4400-bbfe-11eb-8616-46860db1cc44.jpg)

**Comment by iyadahmed** [comment]
> For me it works in the toolshelf.
> 
> 
> 
> ![worksfine](https://user-images.githubusercontent.com/13919702/119270744-6b3a4400-bbfe-11eb-8616-46860db1cc44.jpg)
> 
> 
> 
> 
> 
> 


Not all toolshelf buttons are affected, apply Location, Rotation, etc are affected

**Comment by iyadahmed** [comment]
I didn't see it at first too, but a user reported it

**Comment by ReinerBforartists** [comment]
This is another issue. Which is already in the tracker.

 Separate tooltip for apply transforms #2441

**Comment by ReinerBforartists** [comment]
Same problem here: https://github.com/Bforartists/Bforartists/issues/2437

**Comment by iyadahmed** [comment]
Oh ok my bad then ^^

**Comment by ReinerBforartists** [comment]
No problem. I just know this issues because i had already investigated into it. These two are some very hard cases. No idea how to solve them yet. We might end in separated tooltips in the old python way here.

I had hoped though that your commit will fix the icon only with identical tooltip dilemma in the sidebar. Which is not the case. And so i fear we need to keep this issue open.

![addnodeproblem](https://user-images.githubusercontent.com/13919702/119273385-0802de80-bc0b-11eb-80fe-15a8c9e1ddf1.jpg)

**Comment by iyadahmed** [comment]
> No problem. I just know this issues because i had already investigated into it. These two are some very hard cases. No idea how to solve them yet. We might end in separated tooltips in the old python way here.
> 
> I had hoped though that your commit will fix the icon only with identical tooltip dilemma in the sidebar. Which is not the case. And so i fear we need to keep this issue open. They are all called Add Node ...
> 
> ![addnodeproblem](https://user-images.githubusercontent.com/13919702/119273385-0802de80-bc0b-11eb-80fe-15a8c9e1ddf1.jpg)

Ok, the tooltip (description) itself is not issue, it is the `text` value passed to `layout.operator`, when it is not set or set to empty string (when showing icon only), it shows the operator name, so here' the two ideas:

1. Make `layout.operator` accept `icon_only` parameter, that way we can set text and also show icon only
"layout.prop has this option"
2. Make Operator class accept "label" method or property, for dynamic label, just like it has already "description" method for dynamic tooltip description

Some people asked already about this, but answers assume button is not icon only
https://stackoverflow.com/questions/11508267/how-to-change-operators-label-in-blender-2-63-depending-on-the-context
https://blender.stackexchange.com/questions/81550/dynamically-change-the-bl-label-for-an-operator

**Comment by ReinerBforartists** [comment]
I am honestly still completely lost with this ^^

**Comment by iyadahmed** [comment]
> I am honestly still completely lost with this ^^

No worries I'm working on it now ^^

**Comment by Draise14** [comment]
![bforartists_opxAzedg8n](https://user-images.githubusercontent.com/35357409/224814361-46951da1-15d2-400a-b7dd-c0315c570d12.gif)

![B6WeIIK97N](https://user-images.githubusercontent.com/35357409/224814428-0aa05540-8d3a-478c-a19a-7530e7db9025.gif)

Is this task still relevant? 

In the Node Toolshelf the label "Add Node" is added when in text mode, but the tooltips are showing ok.

**Comment by ReinerBforartists** [comment]
Yes it is. When the title would show as it should then we would not need out tooltip here. And it is not only relevant  in the nodes section. But for me this fish is simply too big.

**Comment by Draise14** [comment]
@iyadahmed I think you are on to something. In theory, if so.. could could I implement the `icon_only` technique you mentioned earlier?

**Comment by ReinerBforartists** [comment]
This taks is now not longer relevant. The tools in the sidebar now all have their title. Closing.

![Image](https://github.com/user-attachments/assets/db4865ce-8f24-41e5-a33b-2577c144b03a)
